stdlib/public/core/CocoaArray.swift

```diff
@@ -47,23 +47,24 @@ internal struct _CocoaArrayWrapper : Collection {
   /// implementation of countByEnumeratingWith.
   func contiguousStorage(
     subRange: Range<Int>
-  ) -> UnsafeMutablePointer<AnyObject>
+  ) -> UnsafeMutablePointer<AnyObject>?
   {
     var enumerationState = _makeSwiftNSFastEnumerationState()
+    var dummyBuffer = [AnyObject]()
 
     // This function currently returns nil unless the first
     // subRange.endIndex items are stored contiguously.  This is an
     // acceptable conservative behavior, but could potentially be
     // optimized for other cases.
     let contiguousCount = withUnsafeMutablePointer(&enumerationState) {
-      self.buffer.countByEnumeratingWith($0, objects: nil, count: 0)
+      self.buffer.countByEnumeratingWith(
+        $0, objects: &dummyBuffer, count: 0)
```

Comment:
So why can't we explicitly deviate from the original form?  It is not like Foundation can change the behavior here -- there are a lot of Swift programs out there that already pass `nil` here.

I just don't see a reason to create more code for the optimizer to work with, and even possibly introducing a performance regression.